spyderlib/baseconfig.py

```diff
@@ -204,18 +205,102 @@ def get_image_path(name, default="not_found.png"):
+    listdir = os.listdir(locale_path)
+    langs = [d for d in listdir if osp.isdir(osp.join(locale_path, d))]
+    langs = [DEFAULT_LANGUAGE] + langs
+
+    # Check that there is a language code available in case a new translation
+    # is added, to ensure LANGUAGE_CODES is updated.
+    for lang in langs:
+        if lang not in LANGUAGE_CODES:
+            error = _('Update LANGUAGE_CODES (inside baseconfig.py) if a new '
+                      'translation has been added to Spyder')
+            raise Exception(error)
+    return langs
+
+
+def get_interface_language():
+    """
+    If Spyder has a translation available for the locale language, it will
+    return the version provided by Spyder adjusted for language subdifferences,
+    otherwise it will return DEFAULT_LANGUAGE.
+
+    Example:
+    1.) Spyder provides ('en',  'fr', 'es' and 'pt_BR'), if the locale is
+    either 'en_US' or 'en' or 'en_UK', this function will return 'en'
+
+    2.) Spyder provides ('en',  'fr', 'es' and 'pt_BR'), if the locale is
+    either 'pt' or 'pt_BR', this function will return 'pt_BR'
+    """
+    locale_language = locale.getdefaultlocale()[0]
+
+    if locale_language is None:
+        language = DEFAULT_LANGUAGE
+    else:
+        spyder_languages = get_available_translations()
+        for spyder_language in spyder_languages:
+            if locale_language == spyder_language:
+                language = locale_language
+                break
+            elif locale_language.startswith(spyder_language) or\
+                    spyder_language.startswith(locale_language):
+                language = spyder_language
+                break
+
+    return language
+
+
+def save_lang_conf(value):
+    """Save language setting to language config file"""
+    with open(LANG_FILE, 'w') as f:
+        f.write(value)
+
+
+def load_lang_conf():
+    """
+    Load language setting from language config file if it exists, otherwise
+    try to use the local settings if Spyder provides a translation, or
+    return the default if no translation provided.
+    """
+    if osp.isfile(LANG_FILE):
+        with open(LANG_FILE, 'r') as f:
+            lang = f.read()
+    else:
+        lang = get_interface_language()
+        save_lang_conf(lang)
+    return lang
+
+
 def get_translation(modname, dirname=None):
     """Return translation callback for module *modname*"""
     if dirname is None:
         dirname = modname
     locale_path = get_module_data_path(dirname, relpath="locale",
                                        attr_name='LOCALEPATH')
-    # fixup environment var LANG in case it's unknown
-    if "LANG" not in os.environ:
-        import locale
-        lang = locale.getdefaultlocale()[0]
-        if lang is not None:
-            os.environ["LANG"] = lang
+
+    # fixup environment var LANG, LANGUAGE
+    language = load_lang_conf()
+    os.environ["LANG"] = language      # Works in windows

```

Comment:
Translations on Mac never have worked because Mac doesn't define proper `locale` values.

This work will let users select the translation they want to use :-)